### PR TITLE
ICU-12811 Replace local-maven-repo with data jar contents

### PR DESCRIPTION
**Pull request by echeran**
It's better to review this PR commit-by-commit, since the addition of the data file contents is very numerous and noisy (and makes Github's UI slow down a lot).

I think [cldr-icu-readme.txt](https://github.com/unicode-org/icu/blob/main/icu4c/source/data/cldr-icu-readme.txt) is the file that contains the stpes for integrating new CLDR data versions into ICU, so it would need to be updated here in this PR, too.

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-12811
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/maven-build/extract-data-files.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Copyright (C) 2016 and later: Unicode, Inc. and others.
```
Comment (mihnita) [inline icu4j/maven-build/extract-data-files.sh:2]: Nitpick: 2023

File: icu4j/maven-build/extract-data-files.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (mihnita) [inline icu4j/maven-build/extract-data-files.sh:1]: Nitpick: to be complete should also have a Windows script.

File: icu4c/source/data/cldr-icu-readme.txt [hunk 3]
```diff
@@ -307,12 +314,24 @@ make icu4j-data-install
 cd $ICU4C_DIR/source/test/testdata
 make icu4j-data-install
 
+# 12c. Replace the extracted {main, test} data files in the Maven build
+
+cd $ICU4J_ROOT/maven-build
+sh extract-data-files.sh
```
Comment (mihnita) [inline icu4c/source/data/cldr-icu-readme.txt:320]: Nitpick :-)
```
./extract-data-files.sh
```

**Review by mihnita — COMMENTED** [review]
Thank you.
Only nitpicks.

But my main worry is the GitHub inability of dealing with many changed files.
I was unable to see the 2 commits that updated `.res` files (even when looking at one commit at the time)
I got an "angry unicorn" page
https://user-images.githubusercontent.com/95055527/200951443-84459209-d082-4f08-ae33-59632b92da30.png

I had to `git clone` your repo locally, `git co also-expand-data-jars`, and review there.
That is quite inconvenient...

File: icu4c/source/data/cldr-icu-readme.txt (hunk 3 above)
Comment (echeran) [reply icu4c/source/data/cldr-icu-readme.txt]: Done.

File: icu4j/maven-build/extract-data-files.sh (hunk 1 above)
Comment (echeran) [reply icu4j/maven-build/extract-data-files.sh]: Done.

File: icu4j/maven-build/extract-data-files.sh (hunk 2 above)
Comment (echeran) [reply icu4j/maven-build/extract-data-files.sh:1]: This script is just to simplify `cldr-icu-readme.txt`, which is a set of steps that need to be run by a person (who can double check discrepencies, etc.), and those steps are written for a POSIX on *nix environment. So I think this is okay for now (until there exists an easy cross-platform solution to do things).

**Comment by echeran** [comment]
> Thank you. Only nitpicks.

Thanks for the review. I made changes to address the comments that were addressable, and comments on the others.

I also pushed a commit to remove more configs (ones that I had missed previously) that are now obsolete after the removal of the local-maven-repo setup.

> But my main worry is the GitHub inability of dealing with many changed files. I was unable to see the 2 commits that updated `.res` files (even when looking at one commit at the time) I got an "angry unicorn" page https://user-images.githubusercontent.com/95055527/200951443-84459209-d082-4f08-ae33-59632b92da30.png
> 
> I had to `git clone` your repo locally, `git co also-expand-data-jars`, and review there. That is quite inconvenient...

All of the non-uncommon CLDR data integration PRs end up like this, too (they're updating all the same files). But for specific purposes of PR reviews, you can use [File Filter drop down](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/filtering-files-in-a-pull-request) to ignore all the binary files if they have a distinct file extension. FWIW, [here is what this PR looks like](https://github.com/unicode-org/icu/pull/2275/files?file-filters%5B%5D=.jar&file-filters%5B%5D=.pom&file-filters%5B%5D=.sh&file-filters%5B%5D=.txt&file-filters%5B%5D=.xml&show-deleted-files=true&show-viewed-files=true) with such filters applied.

**Review by mihnita — APPROVED** [review]

**Comment by srl295** [comment]
@mihnita @echeran late to the party but … it's a good step, but why even have the .res files in the repo? Might consider just rebuilding from the icu4c source similar to https://github.com/unicode-org/icu/pull/164#issuecomment-424872304 - and aggressively caching the result for CI.

**Pull request merged**